### DISCUSSION TITLE
Which brands are Tasmota friendly?

### DISCUSSION THREAD

**Discussion by dac2020** [arendst/Tasmota#16864] [General]
The trouble I'm having is taking a chance buying hardware, never really knowing if I'll be able to convert it to Tasmota in most cases.
I'm wondering if there are some well known brands that are Tasmota friendly, always able to convert.

I see this list but of course, it probably doesn't contain everything that's out there.
https://templates.blakadder.com/preflashed.html

For example, Sonoff is one I seem to see a lot running Tasmota. Is this a brand I can count on to convert? 
I see Athom a lot, is that a Tasmota friendly brand?

**Comment by sfromis** [comment]
There is nothing like some mainline brand exclusively having devices compatible with Tasmota. Obviously, preflashed ones are pretty safe, and Shelly + Sonoff are *mostly* using a compatible MCU, with many Shelly devices flashable without cracking them open, wired or with non-ESP32 devices also mostly OTA.

I've found that Yeelight/Xiaomi has many lamps (not all of them) using ESP32, if you navigate the insides. Some smaller brands promote themselves by stating ESP32 inside as a selling point. Even brands like Gosund which seems to have many ESP devices can be hard to open, meaning not always all that *friendly* to firmware replacement.

On the wider market, the gamble gets much better odds with Zigbee devices suitable for local-only smarthome, without the need for flashing.

**Reply by dac2020** [reply]
Because I'm going to be passing devices on to others, I can't really get into taking them apart or soldering unless it's for my own project or something. This means I have to find stuff that I can flash using wifi or USB or a one time built connector.

I started wondering if there are Ethernet based hardware running Tasmota but then I think Tasmota is mainly all about the WiFi chips.

**Reply by sfromis** [reply]
There are a very few ESP32 devices using the Ethernet feature of that chip, but I've not heard of that providing any avenue to flashing. For regular IoT devices (outside of the breaker panel), Ethernet is just so badly impractical that it is not really a product category.

Shelly devices supported by Tasmota are probably the easiest for wired flashing, when they have a standard 6/7 pin connector accessible from the outside, or are among those where you have an OTA setup for firmware replacement. Arguably, Shelly devices are already so tinker-friendly (including MQTT support) that they are fine for local-only IoT without firmware replacement.

**Reply by dac2020** [reply]
Yes, I found some wifi to Ethernet adapters, maybe called gateways. Just wondered about those because sometimes, you might need an Ethernet connection for some reason or another but I'd like everything to be Tasmota if possible.

**Reply by sfromis** [reply]
I did not mean adapters/gateways, but with ESP32, Tasmota can support both Ethernet and Wifi. There exists relay-type devices with this setup, meaning that it has a socket (RJ-45) for wired Ethernet, usable while running Tasmota, instead of Wifi, or even running both interfaces at the same time.

**Reply by dac2020** [reply]
I've seen those also, where they have both a wifi and Ethernet port.
Can you share one or two so I can keep them in the list I've been creating.

**Reply by sfromis** [reply]
Look here: https://templates.blakadder.com/esp32.html
Several dev boards and gateways, and a Kincony 8-relay board. Shelly has recently released "Shelly Pro" devices with Ethernet, which may get Tasmota support "soon".
